### PR TITLE
Modularization Base Structure

### PR DESCRIPTION
**Pull request by msgharpu**
This commit creates following modules
1) fluentui_core
2) fluentui_listitem
3) fluentui_drawer
4) fluentui_tablayout
5) fluentui_others
6) FluentUI

In addition, contains a build pipeline to create universal package
of the Fluent UI library

### Platforms Impacted
- [*] Android

### Description of changes

This commit creates a base structure for the modularization and adds following modules

1) fluentui_core
2) fluentui_listitem
3) fluentui_drawer
4) fluentui_tablayout

In addition to these, following modules are created

1) fluentui_others - Containing rest of the controls apart from the ones covered above
2) FluentUI - This will act as a container module which has a dependency on all the four modules listed above and fluentui_others

For general users the dependency specification of Fluent UI in gradle will remain the same, though, now, it will fetch five other modules as the dependency

### Verification

Change was tested by verifying all elements in demo app work

### Pull request checklist

This PR has considered:
- [* ] Light and Dark appearances
- [* ] VoiceOver and Keyboard Accessibility
- [*] Internationalization and Right to Left layouts
- [*] Size classes and window sizes (notched devices, multitasking, different window sizes, etc)

### CODE REVIEW

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt [hunk 1]
```diff
@@ -8,10 +8,12 @@ package com.microsoft.fluentuidemo.demos
 import android.os.Bundle
 import android.support.v4.content.ContextCompat
 import android.support.v4.view.PagerAdapter
+import android.util.Log
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:11]: nit

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt [hunk 2]
```diff
@@ -8,10 +8,12 @@ package com.microsoft.fluentuidemo.demos
...
 import android.view.View
 import android.view.ViewGroup
 import com.microsoft.fluentui.tablayout.TabLayout
 import com.microsoft.fluentui.tablayout.TabLayout.TabType.*
+import com.microsoft.fluentui.util.ThemeUtil
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:16]: nit

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml [hunk 3]
```diff
@@ -16,7 +16,7 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         android:padding="@dimen/default_layout_margin"
-        android:background="?attr/fluentuiBackgroundColor"
+        android:background="?attr/fluentuiBottomSheetBackgroundColor"
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml:19]: Why these changes?
I think, we should separate fixes in other PR
Just keep modularize code changes in this PR

File: FluentUI/build.gradle [hunk 4]
```diff
@@ -1,112 +1,90 @@
-/*
- * Copyright (c) Microsoft Corporation. All rights reserved.
- * Licensed under the MIT License.
```
Comment (aishwarya56) [inline FluentUI/build.gradle:3]: nit: Need this copyright statement

File: fluentui_core/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -1,11 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!--
-  ~ Copyright (c) Microsoft Corporation. All rights reserved.
-  ~ Licensed under the MIT License.
```
Comment (aishwarya56) [inline fluentui_core/src/main/res/values/colors.xml:4]: nit: copyright statements

File: fluentui_core/src/androidTest/java/com/microsoft/fluentui/core/ExampleInstrumentedTest.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.microsoft.fluentui.core
```
Comment (aishwarya56) [inline fluentui_core/src/androidTest/java/com/microsoft/fluentui/core/ExampleInstrumentedTest.kt:1]: Please remove these extra test files/directories which are not being used

File: fluentui_core/src/test/java/com/microsoft/fluentui/core/ExampleUnitTest.kt [hunk 7]
```diff
@@ -0,0 +1,17 @@
+package com.microsoft.fluentui.core
```
Comment (aishwarya56) [inline fluentui_core/src/test/java/com/microsoft/fluentui/core/ExampleUnitTest.kt:1]: Extra Files

**Review by aishwarya56 — CHANGES_REQUESTED** [review]
Just an initial Review, please check for some generic comments

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt (hunk 1 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:11]: Corrected

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt (hunk 2 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:16]: Corrected

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml (hunk 3 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml:19]: Corrected. The persistent bottomsheet will have many changes as it has been taken from modular branch which has @mnkshukl changes

File: FluentUI/build.gradle (hunk 4 above)
Comment (msgharpu) [reply FluentUI/build.gradle:3]: Corrected. This has happened due to re-naming of existing files and adding new files for some modules

File: fluentui_core/src/main/res/values/colors.xml (hunk 5 above)
Comment (msgharpu) [reply fluentui_core/src/main/res/values/colors.xml:4]: Corrected

File: fluentui_core/src/androidTest/java/com/microsoft/fluentui/core/ExampleInstrumentedTest.kt (hunk 6 above)
Comment (msgharpu) [reply fluentui_core/src/androidTest/java/com/microsoft/fluentui/core/ExampleInstrumentedTest.kt:1]: Done

File: fluentui_core/src/test/java/com/microsoft/fluentui/core/ExampleUnitTest.kt (hunk 7 above)
Comment (msgharpu) [reply fluentui_core/src/test/java/com/microsoft/fluentui/core/ExampleUnitTest.kt:1]: Done

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml (hunk 3 above)
Comment (aishwarya56) [reply FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml]: That is fine, but we should merge that change separately into develop first. So, that it does not create unwanted changes in this PR.
Request you to get those commits and raise a separate PR

File: fluentui_core/.gitignore [hunk 8]
```diff
@@ -0,0 +1 @@
+/build
```
Comment (aishwarya56) [inline fluentui_core/.gitignore:1]: Should we consider just one .gitignore file, instead of every module having its own?
Not sure of the recommended way. But to check gitignore at multiple locations will be cumbersome

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt [hunk 9]
```diff
@@ -9,7 +9,9 @@ import android.content.Intent
 import android.os.Bundle
 import android.support.annotation.LayoutRes
 import android.support.v7.app.AppCompatActivity
+import android.view.LayoutInflater
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt:12]: Extra Imports,
Please check

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml [hunk 10]
```diff
@@ -40,6 +40,13 @@
             android:layout_marginTop="@dimen/default_layout_margin"
             android:text="@string/toggle_sheet_content" />
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml:42]: Please check the reply to conversation, regarding having a separate PR for these changes related to persistent bottomsheet

File: fluentui_core/src/main/java/com/microsoft/fluentui/view/TemplateView.kt [hunk 11]
```diff
@@ -38,8 +37,7 @@ import com.microsoft.fluentui.theming.FluentUIContextThemeWrapper
  */
 abstract class TemplateView : ViewGroup {
     @JvmOverloads
-    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(FluentUIContextThemeWrapper(context), attrs, defStyleAttr)
```
Comment (aishwarya56) [inline fluentui_core/src/main/java/com/microsoft/fluentui/view/TemplateView.kt:41]: why are we not wrapping here?

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/bottomsheet/BottomSheet.kt [hunk 12]
```diff
@@ -9,6 +9,7 @@ import android.app.Dialog
 import android.content.DialogInterface
 import android.os.Bundle
 import android.support.v7.app.AppCompatDialogFragment
+import java.util.*
```
Comment (aishwarya56) [inline fluentui_drawer/src/main/java/com/microsoft/fluentui/bottomsheet/BottomSheet.kt:12]: avoid using *

File: fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml [hunk 13]
```diff
@@ -28,4 +28,4 @@
             </item>
         </selector>
     </item>
-</ripple>
```
Comment (aishwarya56) [inline fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml:31]: Extra

File: fluentui_core/src/main/java/com/microsoft/fluentui/view/TemplateView.kt (hunk 11 above)
Comment (msgharpu) [reply fluentui_core/src/main/java/com/microsoft/fluentui/view/TemplateView.kt:41]: The context received in constructor is already wrapped with the theme of specific module. E.g for from tablayout we call TemplateView constructor by wrapping in Theme_FluentUI_TabLayout. If we wrap it here again without passing any theme, the default theme is applied (Base_Theme_FluentUI) to the context of say Tablayout, which is not correct

File: fluentui_drawer/src/main/java/com/microsoft/fluentui/bottomsheet/BottomSheet.kt (hunk 12 above)
Comment (msgharpu) [reply fluentui_drawer/src/main/java/com/microsoft/fluentui/bottomsheet/BottomSheet.kt:12]: This is copied as is from modular branch code. Corrected now

File: fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml (hunk 13 above)
Comment (msgharpu) [reply fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml:31]: Do not see this in code. Not sure why showing like this

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml (hunk 3 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml]: It will be cumbersome now  to do that. There were many changes in persistent bottom sheet and as per our earlier discussion, this was taken in to keep it consistent with modular persistent bottomsheet at once. @mnkshukl

File: FluentUI/build.gradle [hunk 14]
```diff
@@ -2,104 +2,81 @@
...
-            pom.withXml {
-                def dependenciesNode = asNode().appendNode('dependencies')
-                // Iterate over the compile dependencies (we don't want the test ones), adding a <dependency> node for each
-                configurations.implementation.allDependencies.each {
-                    if (it.group != null && (it.name != null && it.name != "unspecified") && it.version != null) {
-                        def dependencyNode = dependenciesNode.appendNode('dependency')
-                        dependencyNode.appendNode('groupId', it.group)
-                        dependencyNode.appendNode('artifactId', it.name)
-                        dependencyNode.appendNode('version', it.version)
+project.afterEvaluate {
```
Comment (mnkshukl) [inline FluentUI/build.gradle:52]: whats the purpose of adding afterEvaluate here

File: FluentUI/build.gradle [hunk 15]
```diff
@@ -2,104 +2,81 @@
...
-    implementation 'com.android.support:exifinterface:28.0.0'
-    implementation 'com.android.support:recyclerview-v7:28.0.0'
-    implementation 'com.android.support:cardview-v7:28.0.0'
-    implementation 'com.android.support:design:28.0.0'
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation project(':fluentui_core')
+    implementation project(':fluentui_listitem')
+    implementation project(':fluentui_drawer')
+    implementation project(':fluentui_tablayout')
+    implementation project(':fluentui_others')
```
Comment (mnkshukl) [inline FluentUI/build.gradle:39]: will this make a transient download ?
I think what you showed me was that transient dependency works only with maven depdency nor with project depdency

File: config.gradle [hunk 16]
```diff
@@ -0,0 +1,10 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+project.ext.fluentui_core_versionid='0.0.1'
```
Comment (mnkshukl) [inline config.gradle:5]: can you add here a doc about how versioning would work and when and when not to change this

File: fluentui_core/src/main/java/com/microsoft/fluentui/theming/FluentUIContextThemeWrapper.kt [hunk 17]
```diff
@@ -8,9 +8,9 @@ package com.microsoft.fluentui.theming
 import android.content.Context
 import android.content.res.Resources
 import android.support.v7.view.ContextThemeWrapper
-import com.microsoft.fluentui.R
+import com.microsoft.fluentui.core.R
```
Comment (mnkshukl) [inline fluentui_core/src/main/java/com/microsoft/fluentui/theming/FluentUIContextThemeWrapper.kt:11]: adding manifest package name only changes resource file path, right  ?

File: fluentui_core/src/main/java/com/microsoft/fluentui/view/MSRecyclerView.kt [hunk 18]
```diff
@@ -26,7 +26,7 @@ import android.view.ViewConfiguration
  * Note: The crazy math part has been taken from Android framework Scroller.java and modified
  * to match our needs.
  */
-internal open class MSRecyclerView : RecyclerView {
+open class MSRecyclerView : RecyclerView {
```
Comment (mnkshukl) [inline fluentui_core/src/main/java/com/microsoft/fluentui/view/MSRecyclerView.kt:29]: I am assuming we didn't want to expose this outside library.
If this is done only for cross module use , mark it's scope as library only

File: fluentui_drawer/build.gradle [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+    implementation project(':fluentui_core')
+    implementation project(':fluentui_listitem')
+    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+
+    implementation "com.android.support:appcompat-v7:$supportVersion"
+    implementation "com.android.support:recyclerview-v7:$supportVersion"
+    implementation "com.android.support:design:$supportVersion"
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
```
Comment (mnkshukl) [inline fluentui_drawer/build.gradle:47]: use global variable for versions?

File: fluentui_drawer/build.gradle [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+        publications {
+            FluentUI(MavenPublication) {
+                groupId 'com.microsoft.fluentui'
+                artifactId "fluentui_drawer"
+                version = android.defaultConfig.versionName
+                artifact(sourceJar)
+                artifact(bundleReleaseAar)
+                pom.withXml {
+                    def dependenciesNode = asNode().appendNode('dependencies')
+                    // Iterate over the compile dependencies (we don't want the test ones), adding a <dependency> node for each
```
Comment (mnkshukl) [inline fluentui_drawer/build.gradle:66]: this piece if code looks very redundant ,
can we not use a script from a single place like in core?

File: FluentUI/build.gradle (hunk 15 above)
Comment (msgharpu) [reply FluentUI/build.gradle:39]: This is added here to generate the pom files for bintrayUpload using config.gradle

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt [hunk 21]
```diff
@@ -49,28 +48,29 @@ class TabLayout : TemplateView {
...
+
     @JvmOverloads
-    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(FluentUIContextThemeWrapper(context), attrs, defStyleAttr) {
+    constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : super(FluentUIContextThemeWrapper(context,R.style.Theme_FluentUI_TabLayout), attrs, defStyleAttr) {
         val styledAttributes = context.obtainStyledAttributes(attrs, R.styleable.TabLayout)
         val tabTypeOrdinal = styledAttributes.getInt(R.styleable.TabLayout_tabType, TabType.STANDARD.ordinal)
         containerBackgroundColor = styledAttributes.getColor(
             R.styleable.TabLayout_containerBackgroundColor,
-            ThemeUtil.getColor(context, R.attr.fluentuiTabLayoutContainerBackgroundColor))
+            ThemeUtil.getColor(this.context, R.attr.fluentuiTabLayoutContainerBackgroundColor))
```
Comment (aishwarya56) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt:58]: context that we are getting in params is different from this.context?
Can you please mention why are we using this here and not relying on actual.
you can add comment

File: fluentui_core/build.gradle [hunk 22]
```diff
@@ -0,0 +1,129 @@
...
+    from android.sourceSets.main.java.srcDirs
+    classifier "sources"
+}
+
+
+project.afterEvaluate {
+    publishing {
+
+        publications {
+            FluentUI(MavenPublication) {
```
Comment (aishwarya56) [inline fluentui_core/build.gradle:82]: can we have these at some common place?

File: fluentui_core/.gitignore (hunk 8 above)
Comment (msgharpu) [reply fluentui_core/.gitignore:1]: Done. I'll be removing .gitignore files in all subdirectories in next push

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt (hunk 9 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt:12]: Done. Fixed in next patch

File: FluentUI/build.gradle (hunk 14 above)
Comment (msgharpu) [reply FluentUI/build.gradle:52]: Will be removed in next patch

File: fluentui_core/src/main/java/com/microsoft/fluentui/theming/FluentUIContextThemeWrapper.kt (hunk 17 above)
Comment (msgharpu) [reply fluentui_core/src/main/java/com/microsoft/fluentui/theming/FluentUIContextThemeWrapper.kt:11]: > adding manifest package name only changes resource file path, right ?

Yes

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt (hunk 21 above)
Comment (msgharpu) [reply fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt:58]: As discussed, here the object is not yet initialized fully, so, if we use just context, it takes the local variable context which has theme of the calling application. this.context refers to the context of tablayout object which is wrapped with appropriate theme

File: fluentui_drawer/build.gradle (hunk 20 above)
Comment (msgharpu) [reply fluentui_drawer/build.gradle:66]: The artifact_id is different in every module. If we were to have a possibility where we would publish modules individually going forward, this will be required. But, noted the comment. This will have to be explored in detail. Is it ok if this is taken up later in a separate PR after due analysis?

File: config.gradle (hunk 16 above)
Comment (msgharpu) [reply config.gradle:5]: Done. Will be available in next patch

File: fluentui_core/src/main/java/com/microsoft/fluentui/view/MSRecyclerView.kt (hunk 18 above)
Comment (msgharpu) [reply fluentui_core/src/main/java/com/microsoft/fluentui/view/MSRecyclerView.kt:29]: Done. Will be available in next patch

File: fluentui_drawer/build.gradle (hunk 19 above)
Comment (msgharpu) [reply fluentui_drawer/build.gradle:47]: Done. Will be available in next patch

File: fluentui_core/build.gradle (hunk 22 above)
Comment (msgharpu) [reply fluentui_core/build.gradle:82]: Same as mentioned on Mayank's comment above
https://github.com/microsoft/fluentui-android/pull/41#discussion_r581794316

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml (hunk 3 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml]: @aishwarya56 : As discussed, I compared the files of drawer module from modular branch to the ones in this PR. Most files are same, few changes are with respect to wrapping the context with the drawer theme
Another change is listitem has been moved from core to a separate module

Following is the list of changes 
- BottomSheetItemDivider.kt added themewrapper to constructor
- SheetHorizontalItemList.kt added themewrapper to constructor
- SheetHorizontalItemAdapter.kt passed Drawer theme to constructor
- SheetHorizontalItemView.kt added themewrapper to constructor
           - changed textAppearanceResId to R.style.TextAppearance_FluentUI_HorizontalListItemTitle from R.style.TextAppearance_FluentUI_ListItemSubtitle as listitem is now separate module
 - attrs.xml added fluentuiHorizontalListItemTitleColor
 - styles_font.xml, added TextAppearance.FluentUI.HorizontalListItemTitle
 - themes.xml added fluentuiHorizontalListItemTitleColor 
 - In values-night/themes.xml added section for BottomSheet which was missing in modular branch
 - fluentuiBottomSheetIconColor was added extra in night theme. removed now
 - In ListItemView.kt, added themewrapper to constructor
 - In ListSubHeaderView.kt added themewrapper to constructor

File: FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml (hunk 10 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/res/layout/activity_persistent_bottom_sheet.xml:42]: Replied to the previous conversation

File: fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml (hunk 13 above)
Comment (msgharpu) [reply fluentui_others/src/main/res/drawable/contextual_command_bar_start_item_background.xml:31]: I do not see this in the PR, can we resolve this?

File: fluentui_drawer/build.gradle (hunk 20 above)
Comment (aishwarya56) [reply fluentui_drawer/build.gradle]: @msgharpu request you to create this work item and prioritise it as every new module will have this common code and it will keep on building up on this un-optimized code.
I believe you need to have some common.gradle file and allow it take on some parameters with group and artifect

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt [hunk 23]
```diff
@@ -18,7 +18,6 @@ abstract class DemoActivity : AppCompatActivity() {
     companion object {
         const val DEMO_ID = "demo_id"
     }
-
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt:21]: nit : can we have a last pass too not include extra changes? like line removal

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt [hunk 24]
```diff
@@ -25,7 +25,6 @@ class TabLayoutActivity : DemoActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
```
Comment (aishwarya56) [inline FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:28]: nit

File: FluentUI/build.gradle [hunk 25]
```diff
@@ -2,88 +2,55 @@
...
-    compileSdkVersion 28
+    compileSdkVersion constants.compileSdkVersion
     defaultConfig {
-        minSdkVersion 21
-        targetSdkVersion 28
-        versionCode 11
-        versionName '0.0.11'
+        minSdkVersion constants.minSdkVersion
+        targetSdkVersion constants.targetSdkVersion
+        versionCode 1
```
Comment (aishwarya56) [inline FluentUI/build.gradle:19]: check the version code

File: FluentUI/build.gradle [hunk 26]
```diff
@@ -2,88 +2,55 @@
...
+    androidTestImplementation "com.android.support.test.espresso:espresso-core:$espressoVersion"
 }
-
 task sourceJar(type: Jar) {
     from android.sourceSets.main.java.srcDirs
     classifier "sources"
 }
 
 publishing {
-    repositories {
```
Comment (aishwarya56) [inline FluentUI/build.gradle:78]: Are we not releasing maven now?

File: FluentUI/build.gradle [hunk 27]
```diff
@@ -97,9 +64,17 @@ publishing {
                 configurations.implementation.allDependencies.each {
                     if (it.group != null && (it.name != null && it.name != "unspecified") && it.version != null) {
                         def dependencyNode = dependenciesNode.appendNode('dependency')
-                        dependencyNode.appendNode('groupId', it.group)
-                        dependencyNode.appendNode('artifactId', it.name)
-                        dependencyNode.appendNode('version', it.version)
+                        if (it.group.equals('fluentui-android')) {
```
Comment (aishwarya56) [inline FluentUI/build.gradle:67]: Please explain this if else
You can mention it in the comments

File: FluentUI/src/main/AndroidManifest.xml [hunk 28]
```diff
@@ -3,6 +3,5 @@
   ~ Copyright (c) Microsoft Corporation. All rights reserved.
   ~ Licensed under the MIT License.
   -->
-
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.microsoft.fluentui" />
+    package="com.microsoft.fluentui_container" />
```
Comment (aishwarya56) [inline FluentUI/src/main/AndroidManifest.xml:7]: Is this necessary? container name?
My only concern is package name here says container and in gradle we just say fluentui

File: fluentui_core/src/main/AndroidManifest.xml [hunk 29]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) Microsoft Corporation. All rights reserved.
+  ~ Licensed under the MIT License.
+  -->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.microsoft.fluentui.core" />
```
Comment (aishwarya56) [inline fluentui_core/src/main/AndroidManifest.xml:7]: The package here says com.microsoft.fluentui.core
But in the kotlin filesour package only has a structure till com.microsoft.fluentui
Are we sure it will not cause an issue?
Also the way we refer R is com.microsoft.fluentui.core.R
@mnkshukl your thoughts?

File: fluentui_drawer/src/main/res/layout/view_persistent_sheet.xml [hunk 30]
```diff
@@ -1,11 +1,11 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (aishwarya56) [inline fluentui_drawer/src/main/res/layout/view_persistent_sheet.xml:1]: Please correct move it to next line

File: fluentui_others/src/main/AndroidManifest.xml [hunk 31]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) Microsoft Corporation. All rights reserved.
+  ~ Licensed under the MIT License.
+  -->
+
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.microsoft.fluentui" />
```
Comment (aishwarya56) [inline fluentui_others/src/main/AndroidManifest.xml:8]: same here, this is marked as fluentui but will be released as fluentui_others
but the fluentui.container will be relased as fluentui 
just wanted to know thoughts on this

File: fluentui_others/src/main/res/values-night/themes.xml [hunk 32]
```diff
@@ -74,11 +57,7 @@
         <item name="fluentuiDialogBackgroundColor">?attr/fluentuiBackgroundSecondaryColor</item>
         <item name="fluentuiDialogCloseIconColor">?attr/colorPrimary</item>
 
-        <!--Drawer-->
-        <item name="fluentuiDrawerBackgroundColor">?attr/fluentuiBackgroundSecondaryColor</item>
-        <item name="fluentuiDrawerHandleColor">@color/fluentui_gray_500</item>
-
-        <!--ListItemView-->
+         <!--ListItemView-->
```
Comment (aishwarya56) [inline fluentui_others/src/main/res/values-night/themes.xml:60]: nit:

File: fluentui_others/src/main/res/values/integers.xml [hunk 33]
```diff
@@ -13,15 +13,9 @@
     <integer name="fluentui_anim_indeterminate_duration">1000</integer>
     <integer name="fluentui_anim_indeterminate_linear_duration">2000</integer>
 
-    <!--Drawer-->
-    <integer name="fluentui_drawer_fade_in_milliseconds" type="integer">100</integer>
-    <integer name="fluentui_drawer_fade_out_milliseconds" type="integer">50</integer>
-
-    <!--NumberPicker-->
+   <!--NumberPicker-->
```
Comment (aishwarya56) [inline fluentui_others/src/main/res/values/integers.xml:16]: nit

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt (hunk 21 above)
Comment (aishwarya56) [inline fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt:58]: add proper comments to explain this.context, will be helpful for other people contributing to this

File: FluentUI/build.gradle (hunk 26 above)
Comment (msgharpu) [reply FluentUI/build.gradle:78]: We discussed this offline few days back that no one is using the packages from ADO feed

File: FluentUI/build.gradle (hunk 27 above)
Comment (msgharpu) [reply FluentUI/build.gradle:67]: Will explain it here instead of comments as this code is present in every module.

This code is to generate the pom file for bintray. When reading through the dependency list, whenever a project dependency is encountered, e.g 

implementation project(':fluentui_core')

The groupId is the project name i.e fluentui-android, hence for such cases we explicitly make groupid as com.microsoft.fluentui and read the version id from config.gradle

File: FluentUI/src/main/AndroidManifest.xml (hunk 28 above)
Comment (msgharpu) [reply FluentUI/src/main/AndroidManifest.xml:7]: It can be changed. But, in fluentui_others we have kept package name as com.microsoft.fluentui.
There, its kept as is because fluentui_others represents the controls which are not yet modularized. Hence no code has been changed there. 
Import statement in all files in it still looks like, import com.microsoft.fluentui.R

If we change package name in manifest, this import will have to be changed in all files of fluntui_others.

So, to keep pkg names unique, and as fluentui is used for others, gave this one a different name

File: fluentui_core/src/main/AndroidManifest.xml (hunk 29 above)
Comment (msgharpu) [reply fluentui_core/src/main/AndroidManifest.xml:7]: We have to give one package name to the module so that all common resources are accessible to other packages within the same module - import com.microsoft.fluentui.core.R, hence this is required. @mnkshukl had done this in first modular code for bottomsheet as well

File: fluentui_others/src/main/AndroidManifest.xml (hunk 31 above)
Comment (msgharpu) [reply fluentui_others/src/main/AndroidManifest.xml:8]: Refer to earlier comment : https://github.com/microsoft/fluentui-android/pull/41#discussion_r583870194

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt (hunk 23 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/DemoActivity.kt:21]: Done. Fixed in next patch

File: FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt (hunk 24 above)
Comment (msgharpu) [reply FluentUI.Demo/src/main/java/com/microsoft/fluentuidemo/demos/TabLayoutActivity.kt:28]: Done. Fixed in next patch

File: fluentui_drawer/src/main/res/layout/view_persistent_sheet.xml (hunk 30 above)
Comment (msgharpu) [reply fluentui_drawer/src/main/res/layout/view_persistent_sheet.xml:1]: Done. Fixed in next patch

File: fluentui_others/src/main/res/values-night/themes.xml (hunk 32 above)
Comment (msgharpu) [reply fluentui_others/src/main/res/values-night/themes.xml:60]: Corrected. ListItemView actually should be in listitem module, corrected this

File: fluentui_others/src/main/res/values/integers.xml (hunk 33 above)
Comment (msgharpu) [reply fluentui_others/src/main/res/values/integers.xml:16]: Corrected in next patch

File: fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt (hunk 21 above)
Comment (msgharpu) [reply fluentui_tablayout/src/main/java/com/microsoft/fluentui/tablayout/TabLayout.kt:58]: Done. Fixed in next patch

File: FluentUI/build.gradle (hunk 26 above)
Comment (aishwarya56) [reply FluentUI/build.gradle:78]: I think we should not be removing this and continue publishing on our maven feed.
Biggest reason being Jcenter/bintray will be shutting down soon around May.  Only 2 months from now, many people within Microsoft suggest to use that only

File: FluentUI/src/main/AndroidManifest.xml (hunk 28 above)
Comment (mnkshukl) [reply FluentUI/src/main/AndroidManifest.xml:7]: Just add a TODO to change it to fleuntui after full modularization.

Comment (msgharpu) [reply FluentUI/src/main/AndroidManifest.xml:8]: > you can add a TODO to change it to fleuntui after full modularization.

Done

**Review by mnkshukl — APPROVED** [review]
You can use common gradle file for publication

File: fluentui_others/build.gradle [hunk 34]
```diff
@@ -0,0 +1,131 @@
...
+
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.bintray'
+apply from: '../config.gradle'
+android {
+    compileSdkVersion 28
+    defaultConfig {
+        minSdkVersion 21
+        targetSdkVersion 28
+        versionCode 10
```
Comment (aishwarya56) [inline fluentui_others/build.gradle:19]: please check this

File: fluentui_others/build.gradle [hunk 35]
```diff
@@ -0,0 +1,131 @@
...
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-android-extensions'
+
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.bintray'
+apply from: '../config.gradle'
+android {
+    compileSdkVersion 28
+    defaultConfig {
+        minSdkVersion 21
```
Comment (aishwarya56) [inline fluentui_others/build.gradle:17]: use constants here too

File: config.gradle [hunk 36]
```diff
@@ -0,0 +1,19 @@
...
+ * The version id mentioned here will be used for pom file generation for publication
+ * Whenever a module code is changed the version id of the module and all modules
+ * which are dependent on it should be incremented
+ * E.g If a change is made in fluentui_listitem and its current version is 0.0.2
+ * and fluentui_drawer' current version is 0.0.2, both fluentui_listitem and
+ * fluentui_drawer and FluentUI should increment their respective version ids
+  */
+project.ext.fluentui_core_versionid='0.0.1'
+project.ext.fluentui_listitem_versionid='0.0.1'
+project.ext.fluentui_tablayout_versionid='0.0.1'
```
Comment (aishwarya56) [inline config.gradle:16]: should'nt we use these in gradle files?

**Review by aishwarya56 — APPROVED** [review]
Please do these last changes also take a pull

File: config.gradle (hunk 36 above)
Comment (msgharpu) [reply config.gradle:16]: We use this in gradle files here
--> dependencyNode.appendNode('version', getProperty(artifactName).toString())
here, artifact name is e.g fluentui_tablayout_versionid

**Comment by msgharpu** [comment]
Merging to develop branch
